packages/ember-data/lib/system/model/model.js

```diff
@@ -844,6 +817,35 @@ var Model = Ember.Object.extend(Ember.Evented, {
   },
 
   /**
+    @method _changedKeys
+    @private
+  */
+  _changedKeys: function(updates) {
+    var changedKeys = [];
+
+    if (updates && typeof updates === 'object') {
+      var original, i, value, key;
+      var keys = Ember.keys(updates);
+      var length = keys.length;
+
+      original = merge({}, this._data);
+      original = merge(original, this._attributes);
```

Comment:
not sure this is correct, as if you consider user applying changes after sending to the server, even if server sends a different value, it will not show up once you get it as an attr, as the local change will win